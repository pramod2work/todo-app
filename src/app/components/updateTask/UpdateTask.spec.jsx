/* global describe test expect beforeAll jest */

import React from 'react'
import { shallow } from 'enzyme'

import UpdateTasksComponent from './UpdateTask'

describe('Test for View Tasks Component', () => {
  let component
  const propValues = {
    match: {
      params: { taskId: 'add' }
    },
    tasks: [{
      name: 'test',
      completed: false,
      highPriority: false,
      id: 0,
      key: 0
    }],
    addNewTask: jest.fn(),
    updateTaskById: jest.fn(),
    history: { push: jest.fn() }
  }

  beforeAll(() => {
    component = shallow(<UpdateTasksComponent {...propValues} />)
  })

  test('Check snapshot for add', () => {
    expect(component).toMatchSnapshot()
  })

  test('Check snapshot for update', () => {
    const updateProps = {
      ...propValues,
      match: {
        params: { taskId: 0 }
      }
    }
    const updateComponent = shallow(<UpdateTasksComponent {...updateProps} />)
    expect(updateComponent).toMatchSnapshot()
  })

  test('check for update task name', () => {
    const event = {
      target: {
        type: 'text',
        value: 'Test',
        name: 'name'
      }
    }
    component.instance().handleInputChange(event)
    expect(component.state().name).toBe('Test')
  })

  test('check for complete status change', () => {
    const event = {
      target: {
        type: 'checkbox',
        checked: true,
        name: 'completed'
      }
    }
    component.instance().handleInputChange(event)
    expect(component.state().completed).toBeTruthy()
  })

  test('check for highPriority status change', () => {
    component.setState({
      ...component.state(),
      highPriority: true
    })

    const event = {
      target: {
        type: 'checkbox',
        checked: false,
        name: 'highPriority'
      }
    }
    component.instance().handleInputChange(event)
    expect(component.state().highPriority).toBeFalsy()
  })

  test('check for Invalid Form Submit', () => {
    const event = {
      preventDefault: jest.fn(),
      stopPropagation: jest.fn(),
      currentTarget: {
        checkValidity: jest.fn(() => false)
      }
    }
    component.instance().handleSubmit(event)
    expect(event.preventDefault).toHaveBeenCalled()
    expect(event.stopPropagation).toHaveBeenCalled()
    expect(propValues.updateTaskById).not.toHaveBeenCalled()
    expect(propValues.addNewTask).not.toHaveBeenCalled()
    expect(component.state().validated).toBeTruthy()
  })

  test('check for valid Form Submit for update', () => {
    const updateProps = {
      ...propValues,
      match: {
        params: { taskId: 0 }
      }
    }
    const updateComponent = shallow(<UpdateTasksComponent {...updateProps} />)

    const formData = {
      name: 'Test',
      highPriority: true,
      completed: false
    }
    updateComponent.setState({
      ...updateComponent.state(),
      ...formData,
      taskId: 0
    })
    const event = {
      preventDefault: jest.fn(),
      stopPropagation: jest.fn(),
      currentTarget: {
        checkValidity: jest.fn(() => true)
      }
    }
    updateComponent.instance().handleSubmit(event)

    expect(propValues.updateTaskById).toHaveBeenCalledWith({
      ...formData, id: 0
    })
    expect(propValues.addNewTask).not.toHaveBeenCalled()
  })

  test('check for valid Form Submit for update', () => {
    const formData = {
      name: 'Test',
      highPriority: true,
      completed: false
    }
    component.setState({
      ...component.state(),
      ...formData
    })
    const event = {
      preventDefault: jest.fn(),
      stopPropagation: jest.fn(),
      currentTarget: {
        checkValidity: jest.fn(() => true)
      }
    }
    component.instance().handleSubmit(event)

    expect(propValues.addNewTask).toHaveBeenCalledWith({
      ...formData
    })
  })
})
