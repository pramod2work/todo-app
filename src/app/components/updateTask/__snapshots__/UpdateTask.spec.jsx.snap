// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for View Tasks Component Check snapshot for add 1`] = `
<ForwardRef(Bootstrap(Form))
  noValidate={true}
  onSubmit={[Function]}
  validated={false}
>
  <ForwardRef(Bootstrap(FormGroup))
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      }
    }
    controlId="formHorizontalTask"
  >
    <ForwardRef(ContextTransform)
      column={true}
      sm={2}
    >
      Task
    </ForwardRef(ContextTransform)>
    <ForwardRef(Bootstrap(Col))
      sm={10}
    >
      <ForwardRef(ContextTransform)
        data-test-id="task-name"
        name="name"
        onChange={[Function]}
        placeholder="task"
        required={true}
        type="text"
        value=""
      />
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(FormGroup))>
  <ForwardRef(Bootstrap(FormGroup))
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      }
    }
    controlId="formHorizontalCheck"
  >
    <ForwardRef(Bootstrap(Col))
      sm={
        Object {
          "offset": 2,
          "span": 10,
        }
      }
    >
      <ForwardRef(Bootstrap(FormCheck))
        checked={false}
        data-test-id="task-priority"
        label="High Priority"
        name="highPriority"
        onChange={[Function]}
      />
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(FormGroup))>
  <ForwardRef(Bootstrap(FormGroup))
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      }
    }
    controlId="formHorizontalCompleted"
  >
    <ForwardRef(Bootstrap(Col))
      sm={
        Object {
          "offset": 2,
          "span": 10,
        }
      }
    >
      <ForwardRef(Bootstrap(FormCheck))
        checked={false}
        data-test-id="task-status"
        label="Mark Complete"
        name="completed"
        onChange={[Function]}
      />
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(FormGroup))>
  <ForwardRef(Bootstrap(FormGroup))
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      }
    }
  >
    <ForwardRef(Bootstrap(Col))
      sm={
        Object {
          "offset": 2,
          "span": 10,
        }
      }
    >
      <ForwardRef(Bootstrap(Button))
        data-test-id="submit-action"
        type="submit"
      >
        Add
      </ForwardRef(Bootstrap(Button))>
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(FormGroup))>
</ForwardRef(Bootstrap(Form))>
`;

exports[`Test for View Tasks Component Check snapshot for update 1`] = `
<ForwardRef(Bootstrap(Form))
  noValidate={true}
  onSubmit={[Function]}
  validated={false}
>
  <ForwardRef(Bootstrap(FormGroup))
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      }
    }
    controlId="formHorizontalTask"
  >
    <ForwardRef(ContextTransform)
      column={true}
      sm={2}
    >
      Task
    </ForwardRef(ContextTransform)>
    <ForwardRef(Bootstrap(Col))
      sm={10}
    >
      <ForwardRef(ContextTransform)
        data-test-id="task-name"
        name="name"
        onChange={[Function]}
        placeholder="task"
        required={true}
        type="text"
        value="test"
      />
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(FormGroup))>
  <ForwardRef(Bootstrap(FormGroup))
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      }
    }
    controlId="formHorizontalCheck"
  >
    <ForwardRef(Bootstrap(Col))
      sm={
        Object {
          "offset": 2,
          "span": 10,
        }
      }
    >
      <ForwardRef(Bootstrap(FormCheck))
        checked={false}
        data-test-id="task-priority"
        label="High Priority"
        name="highPriority"
        onChange={[Function]}
      />
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(FormGroup))>
  <ForwardRef(Bootstrap(FormGroup))
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      }
    }
    controlId="formHorizontalCompleted"
  >
    <ForwardRef(Bootstrap(Col))
      sm={
        Object {
          "offset": 2,
          "span": 10,
        }
      }
    >
      <ForwardRef(Bootstrap(FormCheck))
        checked={false}
        data-test-id="task-status"
        label="Mark Complete"
        name="completed"
        onChange={[Function]}
      />
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(FormGroup))>
  <ForwardRef(Bootstrap(FormGroup))
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      }
    }
  >
    <ForwardRef(Bootstrap(Col))
      sm={
        Object {
          "offset": 2,
          "span": 10,
        }
      }
    >
      <ForwardRef(Bootstrap(Button))
        data-test-id="submit-action"
        type="submit"
      >
        Update
      </ForwardRef(Bootstrap(Button))>
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(FormGroup))>
</ForwardRef(Bootstrap(Form))>
`;
